src/v/datalake/translation/deps.cc

```diff
@@ -41,37 +41,31 @@ ss::future<cluster::errc> wait_stm_translated(
 }
 } // namespace
 
-ss::future<> noop_mem_tracker::maybe_reserve_memory(size_t, ss::abort_source&) {
+ss::future<>
+noop_mem_tracker::update_current_memory_usage(size_t, ss::abort_source&) {
     return ss::make_ready_future<>();
 }
-void noop_mem_tracker::update_current_memory_usage(size_t) {}
 void noop_mem_tracker::release() {}
 
-ss::future<> writer_reservations_impl::maybe_reserve_memory(
-  size_t bytes, ss::abort_source& as) {
-    while (_available_memory < bytes) {
+ss::future<> writer_reservations_impl::update_current_memory_usage(
+  size_t new_used_bytes, ss::abort_source& as) {
+    _current_used_bytes = new_used_bytes;
+    // todo: as a potential optimization we can choose to release bytes if the
+    // new usage is less than current
+    while (_current_used_bytes > _reservations.count()) {
+        // reserve any deficit
         auto reservation = co_await _reservations_tracker.reserve_memory(as);
```

Comment:
Not entirely related to this PR, but why not allow for users to make exact-sized reservations rather than needing a new cluster config option `datalake_scheduler_block_size_bytes` to determine some default fixed-size allocation?

Reply:
The idea was to not let fast moving translators (in a tight reservation loop here) starve slow moving translators of memory. A block at a time means a translator is guaranteed atleast that full block even if it is slowly appending things, helps batch things better (there were even some ideas to reserve multiple such blocks per translator, i.e min reservation but none of them materialized).

Reply:
Makes sense, how long do these reservations tend to be held? I assume its until the parquet file is written out by either hitting the max chunk size or lag limit?

Reply:
No, for `datalake_scheduler_time_slice_ms`, that is a single time slice allotted by the scheduler and batching can span multiple such slices, The writer will be forced to flush at the end of slice (which also releases memory).